services/crawler/app/services/indexing_service.py

```
@@ -107,30 +107,58 @@ async def _hash_update(conn: asyncpg.Connection) -> tuple[dict[str, int], asyncp
+        # reintroduced below at embed- and storage-time only, so BM25 on
+        # `chunk_content` keeps matching title/URL keywords until Phase 4
+        # moves that responsibility onto dedicated indexed columns.
+        chunks = chunk_content(filtered)
         if not chunks:
             async with acquire_with_retry(self._pool) as conn:
                 await conn.execute(_UPSERT_WEBSITE_URL, domain, url, title, content_hash, filtered_hash)
             return {"url": url, "status": "empty", "chunks_indexed": 0}
 
-        # Generate embeddings
-        texts = [c.content for c in chunks]
+        # During Phase 1–3 chunk_content carries the "Title\n\nURL\n\n" prefix
```

Ruff (0.15.11)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)